app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -1166,7 +1166,7 @@ define([
         updateBaseImage: function (images, context, isInProductView, eventName) {
             var gallery = context.find(this.options.mediaGallerySelector).data('gallery');
 
-            if (eventName === undefined) {
+            if (eventName === undefined && gallery !== undefined) {
```

Comment:
Hi @niravkrish, thanks for collaboration. Please use `typeof` (typeof gallery === undefined) or cast value to bolean uses !! (!!galery).